internal/core/metadata/rest_addressable_test.go

```diff
@@ -46,15 +46,18 @@ var TestTopic = "TestTopic"
 var TestName = "AddressableName"
 var TestId = "123e4567-e89b-12d3-a456-426655440000"
 
-// ErrorPathParam path parameter value which will trigger the 'mux.Vars' function to throw an error due to the '%' not being followed by a valid hexadecimal number.
+// ErrorPathParam path parameter value which will trigger the 'mux.Vars' function to throw an error due to the '%' not
+// being followed by a valid hexadecimal number.
 var ErrorPathParam = "%zz"
 
-// ErrorPortPathParam path parameter used to trigger an error in the `restGetAddressableByPort` function where the port variable is expected to be a number.
+// ErrorPortPathParam path parameter used to trigger an error in the `restGetAddressableByPort` function where the
+// port variable is expected to be a number.
 var ErrorPortPathParam = "abc"
 
-func TestGetAllAddressables(t *testing.T) {
+// Instantiate one logger mock that is used throughout
+var loggerMock = logger.NewMockClient()
```

Comment:
I'm sorry I didn't communicate my ask clearly.

I didn't intend for you to create a global variable.  I intended for you to create an instance of the logger.NewMockClient() once per test invocation and use that instance wherever a logger is required within that test.

For example, the code that I commented on in rest_addressable_test looked like this before:

``` go
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {

			rr := httptest.NewRecorder()
			restGetAllAddressables(
				rr,
				logger.NewMockClient(),
				tt.dbMock,
				errorconcept.NewErrorHandler(logger.NewMockClient()))
			response := rr.Result()
			if response.StatusCode != tt.expectedStatus {
				t.Errorf("status code mismatch -- expected %v got %v", tt.expectedStatus, response.StatusCode)
				return
			}
		})
	}
```

I was looking for an implementation that looked like this:

``` go
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {

			rr := httptest.NewRecorder()
			loggerMock := logger.NewMockClient()
			restGetAllAddressables(
				rr,
				loggerMock,
				tt.dbMock,
				errorconcept.NewErrorHandler(loggerMock))
			response := rr.Result()
			if response.StatusCode != tt.expectedStatus {
				t.Errorf("status code mismatch -- expected %v got %v", tt.expectedStatus, response.StatusCode)
				return
			}
		})
	}
```

Ask is to remove the global and refactor your tests to instantiate a single logger mock instance per test.

Reply:
Gotcha, I like your approach better, and will make it so. All good, thanks 👍